### PR TITLE
Support Sushi LP tokens

### PR DESCRIPTION
**Pull request by bun919tw**
When users supply their Sushi LP (SLP) into cream lending, the contract will stake their SLP into MasterChef contract to earn sushi rewards. Also, every action (supply, borrow, redeem, repay, and transfer) will trigger the job to claim all the sushi rewards from MasterChef contract and stake them into SushiBar contract. Users could claim their sushi rewards by calling `claimSushi()` function.

### CODE REVIEW

File: contracts/CSLPDelegate.sol [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+        // Update the balance of slpSupplyState.
+        slpSupplyState.balance = sushiBalance();
+    }
+
+    function transferSushi(address user, uint userAccrued) internal returns (uint) {
+        if (userAccrued > 0) {
+            uint sushiRemaining = sushi.balanceOf(address(this));
+            if (userAccrued <= sushiRemaining) {
+                sushi.transfer(user, userAccrued);
```
Comment (0xyolo) [inline contracts/CSLPDelegate.sol:187]: check return?

Comment (bun919tw) [reply contracts/CSLPDelegate.sol:187]: Sushi token will always return true so I think we don't need to check the return value?
```
function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
    _transfer(_msgSender(), recipient, amount);
    return true;
}
```

**Comment by bun919tw** [comment]
Didn't write the scenario since the setup is pretty complicated. However, I use hardhat to write some integration tests for cslp.
https://github.com/CreamFi/comptroller-upgrade/tree/cslp

File: contracts/CSLPDelegate.sol [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        // Approve moving our SLP into the master chef contract.
+        EIP20Interface(underlying).approve(masterChefAddress_, uint(-1));
+
+        initState();
+    }
+
+    /**
+     * @notice Manually claim sushi rewards by user
+     */
+    function claimSushi() public {
```
Comment (Macarse) [inline contracts/CSLPDelegate.sol:73]: wdyt of adding a `claimXSushi()` as well?

**Comment by bun919tw** [comment]
Also, update the tests from here: https://github.com/CreamFi/comptroller-upgrade/tree/cslp

**Comment by bun919tw** [comment]
hardhat test updated: https://github.com/CreamFi/comptroller-upgrade/tree/cslp

**Comment by bun919tw** [comment]
Notice that the last 3 commits are about slimming CToken since the contract size of CSLPDelegate was larger than 24MB.

**Comment by bun919tw** [comment]
Thanks, @Macarse @NIC619 for the review. I'm going to merge this PR.

**Pull request merged**